glean-core/python/glean/_loader.py

```diff
@@ -90,6 +90,44 @@ def __getattr__(self, attr):
         )
 
 
+def event_extra_factory(name, argnames):
+    """
+    Generate a new class, inheriting from `metrics.EventExtraClass`
+    and implementing the `to_ffi_extra` method,
+    which serializes expected attributes to pass over FFI.
+    """
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            if key not in argnames:
+                raise TypeError(
+                    "Argument %s not valid for %s" % (key, self.__class__.__name__)
```

Comment:
Should probably use the more modern `f` strings:
```suggestion
                    f"Argument {key} not valid for {self.__class__.__name__}"
```

Reply:
Hah, that's on me for relying on the internet